src/v/pandaproxy/rest/proxy.cc

```diff
@@ -125,7 +128,10 @@ proxy::proxy(
       _ctx,
       json::serialization_format::application_json)
   , _ensure_started{[this]() { return do_start(); }}
-  , _controller(controller) {}
+  , _controller(controller) {
+    _inflight_config_binding.watch(
+      [this]() { _inflight_sem.set_capacity(_inflight_config_binding()); });
```

Comment:
`_inflight_sem` could be destructed at this point (I don't see another mechanism that unsubscribes the watch, or otherwise sequences the destruction order.